tests/lib/smf/src/test_lib_flat_smf.c

```diff
@@ -102,6 +102,11 @@ static void state_a_run(void *obj)
 	zassert_equal(o->transition_bits, test_value[o->tv_idx],
 		      "Test State A run failed");
 
+	if (o->terminate == RUN) {
+		smf_set_terminate(obj, -1);
+		return;
+	}
+
```

Comment:
I like the original approach better, where each state supported different termination options.  Now you're always forcing state A to terminate for the entry/run/exit actions.

Reply:
fixed